Realm/Realm/Handles/SharedRealmHandle.cs

```diff
@@ -46,6 +46,9 @@ private static class NativeMethods
             [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
             public unsafe delegate void OpenRealmCallback(IntPtr task_completion_source, IntPtr shared_realm, int error_code, byte* message_buf, IntPtr message_len);
 
+            [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+            public delegate void FreeGCHandleCallback(IntPtr handle);
```

Comment:
I know this was the name used previously, but let's rename to something that leaks less implementation details. I'm thinking either `DisposeStateCallback` or `OnBindingContextDestructedCallback`.